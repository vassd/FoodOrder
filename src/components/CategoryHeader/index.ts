export { CategoryHeader } from './CategoryHeader'
export type { Category } from './CategoryHeader'
