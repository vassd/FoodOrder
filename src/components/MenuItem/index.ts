export { MenuItem } from './MenuItem'
export type { MenuItemInterface } from './MenuItem'
