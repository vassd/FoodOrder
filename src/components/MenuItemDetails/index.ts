export { MenuItemDetails } from './MenuItemDetails'
export type { Topping, Crust } from './MenuItemDetails'
