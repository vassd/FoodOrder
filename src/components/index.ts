export { MenuHeader } from './MenuHeader'
export { CategoryHeader } from './CategoryHeader'
export type { Category } from './CategoryHeader'
export { MenuItems } from './MenuItems'
export { MenuItem } from './MenuItem'
export type { MenuItemInterface } from './MenuItem'
