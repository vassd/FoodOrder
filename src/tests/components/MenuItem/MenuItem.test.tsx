import React from 'react'
import '@testing-library/jest-dom'
import { describe, expect, test, vi } from 'vitest'
import { render, screen, fireEvent } from '@testing-library/react'
import { MenuItem } from 'components'

describe('MenuItem test', () => {
  const testName: string = 'pizza'
  const testPrice: number = 10
  const testCurrency: string = 'RON'
  const testIngredients: string[] = ['tomato sauce', 'cheese', 'ham']
  const testCategory: number = 0
  const mockOnClick = vi.fn()

  test('should render correctly', () => {
    render(
      <MenuItem
        name={testName}
        price={testPrice}
        currency={testCurrency}
        ingredients={testIngredients}
        category={testCategory}
        onClick={mockOnClick}
      />
    )

    expect(
      screen.getByText(`(${testIngredients.join(', ')})`)
    ).toBeInTheDocument()
    expect(
      screen.getByText(`${testPrice} ${testCurrency}`)
    ).toBeInTheDocument()
    expect(screen.getByText(testName)).toBeInTheDocument()
  })

  test('should match snapshot', () => {
    const { container } = render(
      <MenuItem
        name={testName}
        price={testPrice}
        currency={testCurrency}
        ingredients={testIngredients}
        category={testCategory}
        onClick={mockOnClick}
      />
    )

    expect(container).toMatchSnapshot()
  })

  test('should call onClick', () => {
    render(
      <MenuItem
        name={testName}
        price={testPrice}
        currency={testCurrency}
        ingredients={testIngredients}
        category={testCategory}
        onClick={mockOnClick}
      />
    )

    const item = screen.getByRole('listitem')
    fireEvent.click(item)
    expect(mockOnClick).toBeCalled()
    expect(mockOnClick).toBeCalledWith('pizza')
  })
})
