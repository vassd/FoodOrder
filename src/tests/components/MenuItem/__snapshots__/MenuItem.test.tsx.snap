// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MenuItem test > should match snapshot 1`] = `
<div>
  <li
    class="MuiListItem-root MuiListItem-gutters css-1nxmd3h-MuiListItem-root"
  >
    <div
      class="MuiGrid-root MuiGrid-container _menu-item_9fa3c5 css-1vam7s3-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-9 _menu-item__name-ingredients-list_9fa3c5 css-1qpzhgb-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-b3ynvt-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <strong>
              pizza
            </strong>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            (tomato sauce, cheese, ham)
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 _menu-item__price_9fa3c5 css-1equabv-MuiGrid-root"
      >
        10 RON
      </div>
    </div>
  </li>
</div>
`;
